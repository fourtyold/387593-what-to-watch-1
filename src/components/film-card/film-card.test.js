import React from "react";
import renderer from "react-test-renderer";
import {HashRouter} from "react-router-dom";
import FilmCard from "./film-card.jsx";

const options = {
  film: {
    id: 1,
    name: `Moonrise Kingdom`,
    page: `https://moonrise.html`,
    genre: `Adventure`,
    previewImage: `https://preview_1.jpg`,
    previewVideoLink: `https://preview_1.mp4`
  },
  leaveHandler: () => {},
  enterHandler: () => {},
  isPlaying: false,
  muted: true
};

function createNodeMock(element) {
  if (element.type === `video`) {
    return {};
  }
  return null;
}

it(`Film card correctly renders`, () => {
  const customMockCreator = {createNodeMock};
  const tree = renderer.create(<HashRouter>
    <FilmCard
      film={options.film}
      onLeave={options.leaveHandler}
      onEnter={options.enterHandler}
      isPlaying={options.isPlaying}
      muted={options.muted}
    />
  </HashRouter>, customMockCreator).toJSON();

  expect(tree).toMatchSnapshot();
});
