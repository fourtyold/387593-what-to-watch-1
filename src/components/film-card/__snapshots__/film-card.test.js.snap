// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film card correctly renders 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div>
    <video />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#/film/1"
      onClick={[Function]}
    >
      Moonrise Kingdom
    </a>
  </h3>
</article>
`;
